code/datums/mood_events/needs_events.dm

```diff
@@ -32,6 +32,16 @@
 	description = "<span class='boldwarning'>Oh god that's disgusting...</span>\n"
 	mood_change = -8
 
+/datum/mood_event/disgust/bad_smell
+	description = "<span class='warning'>You smell something horribly decayed inside this room.</span>\n"
+	mood_change = -3
+	timeout = 1800

```

Comment:
If you want this to only affect people while you are inside of miasma; you should remove the timeout and use SEND_SIGNAL(src, COMSIG_CLEAR_MOOD_EVENT, "nauseating_stench") to remove it if there's not enough miasma.

this is fine too ofcourse but the description makes less sense if you keep the moodlet for 3 minutes after. :)

Reply:
ditto for the other event ofc.